pkg/api/rest/resttest/resttest.go

```diff
@@ -1233,6 +1235,7 @@ func (t *Tester) testWatchFields(obj runtime.Object, emitFn EmitFunc, fieldsPass
 	for _, field := range fieldsPass {
 		for _, action := range actions {
 			options := &api.ListOptions{FieldSelector: field.AsSelector(), ResourceVersion: "1"}
+			ctx = context.WithValue(context.WithValue(ctx, "field", field), "action", action)
```

Comment:
We copy the context in the same way. We want different context. The difference is not judged by object pointer.